src/main/java/com/powsybl/openloadflow/ac/AcloadFlowEngine.java

```diff
@@ -191,7 +192,7 @@ public AcLoadFlowResult run() {
         for (var outerLoopAndContext : Lists.reverse(outerLoopsAndContexts)) {
             var outerLoop = outerLoopAndContext.getLeft();
             var outerLoopContext = outerLoopAndContext.getRight();
-            if (outerLoop instanceof DistributedSlackOuterLoop) {
+            if (outerLoop instanceof DistributedSlackOuterLoop || outerLoop instanceof AreaInterchangeControlOuterloop) {
```

Comment:
We now have two distribution outerloops. 2 is a little small number to generalize but sill I wonder if we should not create an interface (for example IActivePowerDistributionOuterLoop) that would provide:
    A way to identify those loops (since apparently there can be only one) - and simplify this condition
    A method to get the distributedActivePower
    A consistent way to handle the slackDistributionFailureBehavior parameter

Reply:
> We now have two distribution outerloops. 2 is a little small number to generalize but sill I wonder if we should not create an interface (for example IActivePowerDistributionOuterLoop) that would provide: A way to identify those loops (since apparently there can be only one) - and simplify this condition A method to get the distributedActivePower A consistent way to handle the slackDistributionFailureBehavior parameter

this definitely needs to be done, considering that in the near future we would also create an area interchange control outerloop for DC load flow (in DC the problem is trivial when all areas are "closed" meaning all boundaries are accounted, but when some interconnections are not to be considered in the exchange, then an iterative method is needed...
So 2 will become 3, and I would propose to do these refactoring at the time of introduction of the DC AIC outerloop if this is ok for you.

Reply:
That works for me.